### PR TITLE
Added Classic A7A7 platform + optimisations, cleaned up removed hardcoding

### PR DESCRIPTION
**Pull request by swingflip**

### CODE REVIEW

File: libretro/Makefile.libretro [hunk 1]
```diff
@@ -5,7 +5,7 @@ SVN_VERSION := " $(shell cat svn)"
 UNAME=$(shell uname -s)
 UNAMEM=$(shell uname -m)
 
-platform = unix
+#platform = unix
```
Comment (fr500) [inline libretro/Makefile.libretro]: please remove this line completely it shouldn't be there

File: libretro/Makefile.libretro [hunk 2]
```diff
@@ -52,20 +52,42 @@ else ifeq ($(platform),osx)
 	WITH_IPX := 1
 endif
 
-# arm
-ifeq ($(CORTEX_A7), 1)
-	COMMONFLAGS += -marm -mcpu=cortex-a7
+# flags
+
+# Classic Platforms ####################
```
Comment (fr500) [inline libretro/Makefile.libretro]: no need for comments

File: libretro/Makefile.libretro [hunk 3]
```diff
@@ -52,20 +52,42 @@ else ifeq ($(platform),osx)
...
+# Platform affix = classic_<ISA>_<µARCH>
+# Help at https://modmyclassic.com/comp
+
+# (armv7 a7, hard point, neon based) ###
```
Comment (fr500) [inline libretro/Makefile.libretro]: ditto for all of these

File: libretro/libretro.cpp [hunk 4]
```diff
@@ -329,6 +329,8 @@ struct retro_variable vars_advanced[] = {
     { "dosbox_svn_emulated_mouse",        "Gamepad emulated mouse; enable|disable" },
 #if defined(C_DYNREC) || defined(C_DYNAMIC_X86)
     { "dosbox_svn_cpu_core",              "CPU core; auto|dynamic|normal|simple" },
+#elif defined(HAVE_CLASSIC)
```
Comment (fr500) [inline libretro/libretro.cpp]: nope, please remove this part

File: libretro/Makefile.libretro [hunk 5]
```diff
@@ -52,20 +52,42 @@ else ifeq ($(platform),osx)
...
+# flags
+
+# Classic Platforms ####################
+# Platform affix = classic_<ISA>_<µARCH>
+# Help at https://modmyclassic.com/comp
+
+# (armv7 a7, hard point, neon based) ### 
+# NESC, SNESC, C64 mini 
+ifeq ($(platform), classic_armv7_a7)
+	TARGET_NAME  := dosbox
```
Comment (fr500) [inline libretro/Makefile.libretro]: target name is define elsewhere already, please don't do this

File: libretro/Makefile.libretro [hunk 6]
```diff
@@ -52,20 +52,42 @@ else ifeq ($(platform),osx)
...
+	-fdata-sections -ffunction-sections -Wl,--gc-sections \
+	-fno-stack-protector -fno-ident -fomit-frame-pointer \
+	-falign-functions=1 -falign-jumps=1 -falign-loops=1 \
+	-fno-unwind-tables -fno-asynchronous-unwind-tables -fno-unroll-loops \
+	-fmerge-all-constants -fno-math-errno \
+	-marm -mtune=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard
 	WITH_DYNAREC = arm
-	ifeq ($(ARM_NEON), 1)
-		COMMONFLAGS += -mfpu=neon-vfpv4
+	CFLAGS += -DHAVE_CLASSIC
```
Comment (fr500) [inline libretro/Makefile.libretro]: this is not required

File: libretro/Makefile.libretro [hunk 7]
```diff
@@ -52,20 +52,42 @@ else ifeq ($(platform),osx)
 	WITH_IPX := 1
 endif
 
-# arm
-ifeq ($(CORTEX_A7), 1)
```
Comment (fr500) [inline libretro/Makefile.libretro]: don't remove this section, it's used for lakka and friends, create your own section instead and leave this one as is please.

**Comment by swingflip** [comment]
Made adjustments as requested @fr500

**Pull request merged**